docs/dogstatsd/configuration.md

```diff
@@ -0,0 +1,144 @@
+# DogStatsD
+
+Information on DogStatsD, configuration and troubleshooting is available in the Datagdog documentation:
+[https://docs.datadoghq.com/developers/dogstatsd/][1]
+
+**The documentation available in this directory is intended for Dogstatsd developers.**
+
+[1]: https://docs.datadoghq.com/developers/dogstatsd/
+
+# Configuration fields
+
+## `dogstatsd_buffer_size`
+
+The `dogstatsd_buffer_size` parameter configures two things in Dogstatsd:
+
+* How many bytes must be read each time the socket is read
+* How many bytes maximum the PacketAssembler should put into one packet
+
+If you have reports of malformed or incomplete packets received by the Dogstatsd server, it could mean
+that the clients that are sending packets are larger than the size of this buffer. If the maximum size of the
+packets sent by the clients can't be changed, consider increasing the size of `dogstatsd_buffer_size`
+as a fallback.
+
+The default value of this field is `8192`.
+
+## `dogstatsd_packet_buffer_size` and `dogstatsd_packet_buffer_flush_timeout`
+
+**Note: This is an internal configuration field subject to change (or to removal) without any notice.**
+
+In order to batch the parsing of the metrics instead of parsing them one after the other every time
+a metric is received, Dogstatsd is using a PacketsBuffer to group packets together.
+
+This configuration field represents how many packets the packets buffer is batching. Decreasing the
+size of this buffer will result in the packets buffer flushing more often the packets to the parser.
+It will be most costly in CPU but could help stress the pipeline and have fewer packet drops when packet
+drops are an issue.
+
+The default value of `dogstatsd_packet_buffer_size` is `32`.
+
+In the same manner, `dogstatsd_packet_buffer_flush_timeout` is a timer forcing a flush on the packets
+buffer to the parser. Each time this timer is triggered, a flush is executed.
+
+The default value of this timer is `100ms`, decreasing it to something small such as `20ms` or even `1ms`
+will use more CPU, but will again stress the pipeline and should lead to fewer drops when packet drops
+are an issue.
+
+## dogstatsd_queue_size
+
+**Note: This is an internal configuration field subject to change (or to removal) without any notice.**
```

Comment:
I guess we can remove this note since it'll now be documented here: https://github.com/DataDog/documentation/pull/7248